CMakeLists.txt

```diff
@@ -46,6 +47,35 @@ OPTION(FIX_RPATH "Correctly set rpath for the linker" OFF)
 OPTION(DEFAULT_COLUMN_MAJOR "Set xtensor default layout to column major. This is currently not supported, since FFTW demands row major layout." OFF)
 OPTION(COVERAGE "Enable coverage compile flags (gcc only!)" OFF)
 OPTION(DISABLE_EXCEPTIONS "Disable C++ exceptions" OFF)
+OPTION(FFTW_USE_FLOAT "Enable FFTW Float type" ON)
+OPTION(FFTW_USE_DOUBLE "Enable FFTW Double type" ON)
+OPTION(FFTW_USE_LONG_DOUBLE "Enable FFTW Long Double type" ON)
+
+if(FFTW_USE_FLOAT)
+    add_definitions(-DXTENSOR_FFTW_USE_FLOAT)
```

Comment:
At this cmake stage, the `target` does not yet exist. That's why I create a definition that can be used on all submodules.
Please, can you explain me the idea about `xtensor-fftw::double` and how use it ? Do you have some project example to show me that use this approach ? Thanks

Reply:
The idea is to have some extra targets that include the correct definitions. I did it in xtensor here: https://github.com/xtensor-stack/xtensor/blob/b5c9286bdce00fa8fc2a93d94303388741624607/xtensorConfig.cmake.in#L54

Reply:
It could be quite nice to have, then you have a bunch of targets and you can hand-pick them when needed. It would avoid to set some options before `find_package`

Reply:
Yes, it applies to the whole cmake tree instead of the given target. And I'm pretty sure it is not exported in the cmake "package" files.

Thus my remark on modernizing the CMake, but this can be done in a dedicated PR.